aws/resource_aws_dx_transit_virtual_interface.go

```diff
@@ -0,0 +1,228 @@
+				Required: true,
+				ForceNew: true,
+			},
+			"customer_address": {
+				Type:     schema.TypeString,
+				Optional: true,
+				Computed: true,
+				ForceNew: true,
+			},
+			"dx_gateway_id": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+			"jumbo_frame_capable": {
+				Type:     schema.TypeBool,
+				Computed: true,
+			},
+			"mtu": {
+				Type:         schema.TypeInt,
+				Default:      1500,
+				Optional:     true,
+				ValidateFunc: validation.IntInSlice([]int{1500, 8500}),
+			},
+			"name": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+			"tags": tagsSchema(),
+			"vlan": {
+				Type:         schema.TypeInt,
+				Required:     true,
+				ForceNew:     true,
+				ValidateFunc: validation.IntBetween(1, 4094),
+			},
+		},
+
+		Timeouts: &schema.ResourceTimeout{
+			Create: schema.DefaultTimeout(10 * time.Minute),
+			Update: schema.DefaultTimeout(10 * time.Minute),
+			Delete: schema.DefaultTimeout(10 * time.Minute),
+		},
+	}
+}
+
+func resourceAwsDxTransitVirtualInterfaceCreate(d *schema.ResourceData, meta interface{}) error {
+	conn := meta.(*AWSClient).dxconn
+
+	req := &directconnect.CreateTransitVirtualInterfaceInput{
+		ConnectionId: aws.String(d.Get("connection_id").(string)),
+		NewTransitVirtualInterface: &directconnect.NewTransitVirtualInterface{
+			AddressFamily:          aws.String(d.Get("address_family").(string)),
+			Asn:                    aws.Int64(int64(d.Get("bgp_asn").(int))),
+			DirectConnectGatewayId: aws.String(d.Get("dx_gateway_id").(string)),
+			Mtu:                    aws.Int64(int64(d.Get("mtu").(int))),
+			VirtualInterfaceName:   aws.String(d.Get("name").(string)),
+			Vlan:                   aws.Int64(int64(d.Get("vlan").(int))),
+		},
+	}
+	if v, ok := d.GetOk("amazon_address"); ok && v.(string) != "" {
+		req.NewTransitVirtualInterface.AmazonAddress = aws.String(v.(string))
+	}
+	if v, ok := d.GetOk("bgp_auth_key"); ok {
+		req.NewTransitVirtualInterface.AuthKey = aws.String(v.(string))
+	}
+	if v, ok := d.GetOk("customer_address"); ok && v.(string) != "" {
+		req.NewTransitVirtualInterface.CustomerAddress = aws.String(v.(string))
+	}
+
+	log.Printf("[DEBUG] Creating Direct Connect transit virtual interface: %#v", req)
+	resp, err := conn.CreateTransitVirtualInterface(req)
+	if err != nil {
+		return fmt.Errorf("error creating Direct Connect transit virtual interface: %s", err)
+	}
+
+	d.SetId(aws.StringValue(resp.VirtualInterface.VirtualInterfaceId))
+	arn := arn.ARN{
+		Partition: meta.(*AWSClient).partition,
+		Region:    meta.(*AWSClient).region,
+		Service:   "directconnect",
+		AccountID: meta.(*AWSClient).accountid,
+		Resource:  fmt.Sprintf("dxvif/%s", d.Id()),
+	}.String()
+	d.Set("arn", arn)
+
+	if err := dxTransitVirtualInterfaceWaitUntilAvailable(conn, d.Id(), d.Timeout(schema.TimeoutCreate)); err != nil {
+		return err
+	}
+
+	return resourceAwsDxTransitVirtualInterfaceUpdate(d, meta)
```

Comment:
Since `CreateTransitVirtualInterface` can support tag-on-create and already handles MTU appropriately above, we should prefer to call `Read` after `Create`.

```suggestion
	return resourceAwsDxTransitVirtualInterfaceRead(d, meta)
```

Reply:
Will do.